SignalMessaging/utils/OWSAvatarBuilder.m

```diff
@@ -50,34 +51,194 @@ + (UIImage *)buildRandomAvatarWithDiameter:(NSUInteger)diameter
     NSString *randomEyebrow = eyebrows[arc4random_uniform((uint32_t)eyebrows.count)];
     NSString *face = [NSString stringWithFormat:@"%@%@%@", randomEyebrow, randomEye, randomMouth];
 
-    CGFloat fontSize = (CGFloat)(diameter / 2.4);
+    UIColor *backgroundColor = [UIColor colorWithRGBHex:0xaca6633];
+
+    return [self avatarImageWithDiameter:diameter
+                         backgroundColor:backgroundColor
+                               drawBlock:^(CGContextRef context) {
+                                   CGContextTranslateCTM(context, diameter / 2, diameter / 2);
```

Comment:
heh, I don't know if it's worth abstracting this to yield the drawing context if it's only used by this debug code.

Admittedly the old debug code was ugly, but I prefer ugly debug code vs. uglifying the production code to make the debug code look nicer.

Reply:
This isn't just to DRY up the debug code, I'm also working towards DRYing up with the group avatars. We're missing an asset but when that's delivered we'll also use this there.